presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -614,4 +626,40 @@ public TableHandle getTarget()
             return target;
         }
     }
+
+    public static final class JoinUsingAnalysis
+    {
+        private final List<Integer> leftJoinFields;
+        private final List<Integer> rightJoinFields;
+        private final List<Integer> otherLeftFields;
+        private final List<Integer> otherRightFields;
+
+        public JoinUsingAnalysis(List<Integer> leftJoinFields, List<Integer> rightJoinFields, List<Integer> otherLeftFields, List<Integer> otherRightFields)
+        {
+            this.leftJoinFields = leftJoinFields;
```

Comment:
`requireNonNull` and `Immutable.copyOf`?

Reply:
Can you please add `checkArgument` that `leftJoinFields` size is equal to `rightJoinFields`?